File: backend/member-service/src/main/java/com/example/memberservice/member/dto/UpdateProfileDto.java
package com.example.memberservice.member.dto;

import com.example.memberservice.profile.dto.ProfileDto;
import com.example.memberservice.profile.dto.ProfileType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UpdateProfileDto {

    private String username;
    private String statusMessage;
    private String profileImage;
    private String wallpaperImage;

    public static UpdateProfileDto createUpdateProfileDto(MemberDto memberDto, ProfileDto profileDto) {
        String profileImage = profileDto.getProfileType().equals(ProfileType.PROFILE_IMAGE) ? profileDto.getValue() : memberDto.getProfileImage();
        String wallpaperImage = profileDto.getProfileType().equals(ProfileType.PROFILE_WALLPAPER) ? profileDto.getValue() : memberDto.getWallpaperImage();

        return UpdateProfileDto.builder()
                .username(memberDto.getUsername())
                .statusMessage(memberDto.getStatusMessage())
                .profileImage(profileImage)
                .wallpaperImage(wallpaperImage)
                .build();
    }
}

File: backend/member-service/src/main/kotlin/com/example/memberservice/member/dto/UpdateProfileDto.kt
package com.example.memberservice.member.dto

import com.example.memberservice.profile.dto.ProfileDto
import com.example.memberservice.profile.dto.ProfileType

data class UpdateProfileDto(
    var username: String? = null,
    var statusMessage: String? = null,
    var profileImage: String? = null,
    var wallpaperImage: String? = null,
) {
    companion object {
        @JvmStatic
        fun createUpdateProfileDto(memberDto: MemberDto, profileDto: ProfileDto): UpdateProfileDto {
            val profileImage = if (profileDto.profileType == ProfileType.PROFILE_IMAGE) profileDto.value else memberDto.profileImage
            val wallpaperImage = if (profileDto.profileType == ProfileType.PROFILE_WALLPAPER) profileDto.value else memberDto.wallpaperImage

            return UpdateProfileDto(
                username = memberDto.username,
                statusMessage = memberDto.statusMessage,
                profileImage = profileImage,
                wallpaperImage = wallpaperImage,
            )
        }
    }
}
